app/api/v2/identity/sessions.rb

```diff
@@ -111,29 +117,111 @@ def get_user(email)
                    desc: 'ID Token'
         end
         post '/auth0' do
-          begin
-            # Decode ID token to get user info
-            claims = Barong::Auth0::JWT.verify(params[:id_token]).first
-            error!({ errors: ['identity.session.auth0.invalid_params'] }, 401) unless claims.key?('email')
-            user = User.find_by(email: claims['email'])
-
-            # If there is no user in platform and user email verified from id_token
-            # system will create user
-            if user.blank? && claims['email_verified']
-              user = User.create!(email: claims['email'], state: 'active')
-              user.labels.create!(scope: 'private', key: 'email', value: 'verified')
-            elsif claims['email_verified'] == false
-              error!({ errors: ['identity.session.auth0.invalid_params'] }, 401) unless user
+          # Decode ID token to get user info
+          claims = Barong::Auth0::JWT.verify(params[:id_token]).first
+          error!({ errors: ['identity.session.auth0.invalid_params'] }, 401) unless claims.key?('email')
+          user = User.find_by(email: claims['email'])
+
+          # If there is no user in platform and user email verified from id_token
+          # system will create user
+          if user.blank? && claims['email_verified']
+            user = User.create!(email: claims['email'], state: 'active')
+            user.labels.create!(scope: 'private', key: 'email', value: 'verified')
+          elsif claims['email_verified'] == false
+            error!({ errors: ['identity.session.auth0.invalid_params'] }, 401) unless user
+          end
+
+          activity_record(user: user.id, action: 'login', result: 'succeed', topic: 'session')
+          csrf_token = open_session(user)
+          publish_session_create(user)
+
+          present user, with: API::V2::Entities::UserWithFullInfo, csrf_token: csrf_token
+        rescue StandardError => e
+          report_exception(e)
+          error!({ errors: ['identity.session.auth0.invalid_params'] }, 422)
+        end
+
+        namespace :switch do
+          desc 'Switch user session',
+               success: { code: 200, message: 'Session was switched' },
+               failure: [
+                 { code: 404, message: 'Record is not found' }
+               ]
+          params do
+            optional :oid,
+                     type: String,
+                     desc: 'Organization OID'
+            optional :uid,
+                     type: String,
+                     desc: 'User UID'
+          end
+          post do
+            user = User.find_by_uid(user_uid)
+            error!({ errors: ['identity.session.not_found'] }, 404) unless user
+
+            switch_oid = params[:oid]
+            # Check switch session mode. Switch proceed only if params[:oid] provided
+            unless switch_oid.nil?
```

Comment:
better to stop the execution before if param is nil

Reply:
params[:oid] is allow nil. If it's nil means. You need to switch back (destroy) as individual user.

Reply:
then can you handle this case first?
it's better to avoid negative conditions to make the code more readable.